src/os/kstore/KStore.cc

```diff
@@ -526,7 +531,9 @@ int KStore::OnodeHashLRU::trim(int max)
   dout(20) << __func__ << " max " << max
 	   << " size " << onode_map.size() << dendl;
   int trimmed = 0;
-  int num = onode_map.size() - max;
+
+  
+  int num = onode_map.size() - std::max(max, 0);
```

Comment:
`trim(max) ` with default value -1. `num` will be onode_map 's size + 1. kstore segfault in https://github.com/ceph/ceph/pull/32446/files#diff-c0f9d6ac22355875cdbbce2f916350d1L549